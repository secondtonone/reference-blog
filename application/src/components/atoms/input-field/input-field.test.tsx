import renderer from 'react-test-renderer';
import InputField from './input-field';

jest.mock('../tooltip', () => ({ children }) => <span>{children}</span>);
const testHandler = () => ({});

describe('Match snapshots InputField', () => {
  it('empty field', () => {
    const tree = renderer.create(<InputField placeholder="Field" value="" onChange={testHandler} scrollTooltip={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with value', () => {
    const tree = renderer.create(<InputField value="Value" onChange={testHandler} scrollTooltip={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('internal', () => {
    const tree = renderer.create(<InputField value="Value" internal onChange={testHandler} scrollTooltip={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with error', () => {
    const tree = renderer.create(<InputField value="Value" onChange={testHandler} state="invalid" error="Error" scrollTooltip={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
