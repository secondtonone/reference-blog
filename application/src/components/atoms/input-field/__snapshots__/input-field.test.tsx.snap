// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots InputField empty field 1`] = `
<span
  className="field-container__Container-sc-qv1os7-0 jOqnRO"
  data-test="field-container"
>
  <span>
    <div
      className="___SInput_ig413_gg_ _size_xl_ig413_gg_ _state_normal_ig413_gg_"
      data-ui-name="Input"
      size="xl"
      style={Object {}}
    >
      <input
        className="___SValue_ig413_gg_ _size_xl_ig413_gg_"
        data-ui-name="Input.Value"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Field"
        size="xl"
        style={Object {}}
        value=""
      />
      <div
        className="___SOutline_ig413_gg_"
      />
    </div>
  </span>
</span>
`;

exports[`Match snapshots InputField internal 1`] = `
<span
  className="field-container__Container-sc-qv1os7-0 trAty"
  data-test="field-container"
>
  <span>
    <div
      className="___SInput_ig413_gg_ _size_xl_ig413_gg_ _state_normal_ig413_gg_"
      data-ui-name="Input"
      size="xl"
      style={Object {}}
    >
      <input
        className="___SValue_ig413_gg_ _size_xl_ig413_gg_"
        data-ui-name="Input.Value"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        size="xl"
        style={Object {}}
        value="Value"
      />
      <div
        className="___SOutline_ig413_gg_"
      />
    </div>
  </span>
</span>
`;

exports[`Match snapshots InputField with error 1`] = `
<span
  className="field-container__Container-sc-qv1os7-0 jOqnRO"
  data-test="field-container"
>
  <span>
    <div
      className="___SInput_ig413_gg_ _size_xl_ig413_gg_ _state_invalid_ig413_gg_"
      data-ui-name="Input"
      size="xl"
      style={Object {}}
    >
      <input
        className="___SValue_ig413_gg_ _size_xl_ig413_gg_"
        data-ui-name="Input.Value"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        size="xl"
        style={Object {}}
        value="Value"
      />
      <div
        className="___SOutline_ig413_gg_"
      />
    </div>
  </span>
</span>
`;

exports[`Match snapshots InputField with value 1`] = `
<span
  className="field-container__Container-sc-qv1os7-0 jOqnRO"
  data-test="field-container"
>
  <span>
    <div
      className="___SInput_ig413_gg_ _size_xl_ig413_gg_ _state_normal_ig413_gg_"
      data-ui-name="Input"
      size="xl"
      style={Object {}}
    >
      <input
        className="___SValue_ig413_gg_ _size_xl_ig413_gg_"
        data-ui-name="Input.Value"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        size="xl"
        style={Object {}}
        value="Value"
      />
      <div
        className="___SOutline_ig413_gg_"
      />
    </div>
  </span>
</span>
`;
